pkg/utils/retry_test.go

```diff
@@ -313,3 +314,82 @@ func TestTryUntil(t *testing.T) {
 	}
 
 }
+
+type TestRetryErrorWithContextItem struct {
+	doc           string
+	timeout       time.Duration
+	workOnTry     int
+	expectedTries int
+	expectSuccess bool
+}
+
+func TestRetryErrorWithContext(t *testing.T) {
+	testTable := []TestRetryErrorWithContextItem{
+		{
+			doc:           "The execution should work at the first try",
```

Comment:
is this a test where this text will never be viewed by ordinary users?

Reply:
Yes, it is.